homeassistant/helpers/event.py

```diff
@@ -774,16 +773,62 @@ def async_refresh(self) -> None:
         """Force recalculate the template."""
         self._refresh(None)
 
-    @callback
-    def _event_triggers_template(self, template: Template, event: Event) -> bool:
```

Comment:
Doesn't it make more sense to keep this in `event.py`, as it's related to tracking results (and references events) ?

Also, each time I see this code I need to look twice to figure out the logic. Could we add parentheses:

```python
        return (
            self._info[template].filter(entity_id)
            or event.data.get("new_state") is None
            or (
                event.data.get("old_state") is None
                and self._info[template].filter_lifecycle(entity_id)
            )
        )
```

Reply:
I debated if this belonged in template or event and landed it in template since I was on the fence.  I'll move it back to event.py and change the syntax so its easier to follow.